### PR TITLE
Added Table of Contents to Readme.md

### PR DESCRIPTION
**Pull request by jobrios**

### CODE REVIEW

**Comment by brendanashworth** [comment]
Can this please be merged? The docs are a pain in the ass to traverse right now.

**Comment by dougwilson** [comment]
Do we know if this works on the new npm website?

**Comment by jobrios** [comment]
As an example, Bootstrap's [Readme.md](https://github.com/twbs/bootstrap/blob/master/README.md) on github is displayed properly on [their npmjs page](https://www.npmjs.com/package/bootstrap).

**Comment by dougwilson** [comment]
Right, but I mean, npm is completely different now, so just need someone (or me) to verify that all the anchor links in the pr work on the new npm.

File: Readme.md [hunk 1]
```diff
@@ -6,6 +6,49 @@
...
+- [Connection options](#connection-options)
+- [SSL options](#ssl-options)
+- [Terminating connections](#terminating-connections)
+- [Pooling connections](#pooling-connections)
+- [Pool options](#pool-options)
+- [Pool events](#pool-events)
+- [Closing all the connections in a pool](#closing-all-the-connections-in-a-pool)
+- [PoolCluster](#poolcluster)
+- [PoolCluster Option](#poolcluster-option)
+- [Switching users / altering connection state](#switching-users--altering-connection-state)
```
Comment (dougwilson) [inline Readme.md]: This link doesn't work on the npmjs.com site.

**Comment by dougwilson** [comment]
Sorry, I reverted your PR merge; I need your name and email address in order to merge your contribution.

**Comment by jobrios** [comment]
Ok. You can see the "Switching users and altering connection state" and "Piping results with Streams2" links working correctly on [my practice package page](https://www.npmjs.com/package/jobrios-npm-practice). If this is ok, I'll go ahead and commit.

Small change: I moved the Streams2 link from the "Piping results with Streams2" section heading to the single occurrence of "Streams2" in the section's text. Otherwise, the Markdown-HTML converter's avoidance of putting a link within a link leads to the section heading not rendering correctly as an anchor link (i.e., no little anchor icon appears on hover), even if you try
```
 ### <a name="piping-results-with-streams2"></a>Piping results with [Streams2](http://blog.nodejs.org/2012/12/20/streams2/)
```

**Comment by dougwilson** [comment]
That all sounds good. I don't know why there is a link in the header in the first place.

**Comment by jobrios** [comment]
Fixed.

**Comment by dougwilson** [comment]
@jobrios unless you provide the information from https://github.com/felixge/node-mysql/pull/930#issuecomment-73898231 I won't be able to merge this as your contribution, if at all.

**Comment by jobrios** [comment]
Ok. The information is now displayed on [my profile page](https://github.com/jobrios).

**Pull request closed without merging**